build.gradle

```diff
@@ -70,6 +70,52 @@ dependencies {
     implementation 'com.github.luben:zstd-jni:1.5.5-4'
     implementation 'org.lz4:lz4-java:1.8.0'
     implementation 'org.xerial.snappy:snappy-java:1.1.10.5'
+    implementation 'software.amazon.msk:aws-msk-iam-auth:2.2.0'
+    // transitive deps used by software.amazon.msk:aws-msk-iam-auth
```

Comment:
😳 

This seems like... a lot. Are we concerned that we are going to have issues with dependency clashes with other plugins and Logstash core? 

cc @jsvd

Reply:
In 4c37ca950c911ee87d837b51115127e5d6afdb21 you can see the difference switching from `1.1.4` to `2.2.0`. While there is a consistent part of AWS libraries (part of groups `software.amazon.awssdk` and `com.amazonaws`) with this new version it adds:
- `commons-codec:commons-codec`
- `commons-logging:commons-logging`
- `io.netty:netty-*`
- `org.reactivestreams:reactive-streams`
- `org.apache.httpcomponents` `httpclient` and `httpcore`

While Netty http and http2 codecs are present it was added also a dependency on Apache's `httpclient` so seems redundant.
The risky one is Netty, because used in other Logstash plugins.

Reply:
From Logstash core we have:
- `httpcore-4.4.16.jar` vs this `4.4.13` (no apparent conflict)
- `httpclient-4.5.14.jar` vs this `4.5.13` (no apparent conflict)
- `commons-codec-1.17.0.jar` vs this `1.15` API **should be compatible**
- `commons-logging-1.3.1.jar` vs this `1.2` API **should be compatible**
- Netty in [Beats](https://github.com/logstash-plugins/logstash-input-beats/blob/870c7431eb28723088c542e8df30e27ff380407a/build.gradle#L22)/[TCP](https://github.com/logstash-plugins/logstash-input-tcp/blob/16d987bb39e7373899f437bb44176c4529dca4bd/build.gradle#L49)/[HTTP](https://github.com/logstash-plugins/logstash-input-http/blob/aba42b3a5867325ba3b7731de23ba754c5475864/build.gradle#L16) inputs are at `4.1.109` vs this `4.1.108` so not so far and **shouldn't break**

Reply:
The pluggable security classes root in 2 classes:
- `software.amazon.msk.auth.iam.IAMLoginModule`
- `software.amazon.msk.auth.iam.IAMClientCallbackHandler`

The `IAMClientCallbackHandler` explicitly instantiates `MSKCredentialProvider` at: https://github.com/aws/aws-msk-iam-auth/blob/v2.2.0/src/main/java/software/amazon/msk/auth/iam/IAMClientCallbackHandler.java#L54

The `MSKCredentialProvider` has direct dependency on 3 main AWSSDK packages:
- `software.amazon.awssdk.auth` provided by `software.amazon.awssdk:auth`
- `software.amazon.awssdk.core` provided by `software.amazon.awssdk:sdk-core`
- `software.amazon.awssdk.services.sts` provide by `software.amazon.awssdk:sts`

Looking at the source of `aws-msk-iam-auth` the artifact `software.amazon.awssdk:sdk-core` requires the following dependency tree:
```
+--- software.amazon.awssdk:sdk-core:2.26.8
    +--- software.amazon.awssdk:annotations:2.26.8
    +--- software.amazon.awssdk:http-client-spi:2.26.8
    |    +--- software.amazon.awssdk:annotations:2.26.8
    |    +--- software.amazon.awssdk:utils:2.26.8 (*)
    |    +--- software.amazon.awssdk:metrics-spi:2.26.8
    |    |    +--- software.amazon.awssdk:annotations:2.26.8
    |    |    \--- software.amazon.awssdk:utils:2.26.8 (*)
    |    \--- org.reactivestreams:reactive-streams:1.0.4
    +--- software.amazon.awssdk:metrics-spi:2.26.8 (*)
    +--- software.amazon.awssdk:endpoints-spi:2.26.8
    |    \--- software.amazon.awssdk:annotations:2.26.8
    +--- software.amazon.awssdk:http-auth-spi:2.26.8
    |    +--- software.amazon.awssdk:annotations:2.26.8
    |    +--- software.amazon.awssdk:utils:2.26.8 (*)
    |    +--- software.amazon.awssdk:http-client-spi:2.26.8 (*)
    |    +--- org.reactivestreams:reactive-streams:1.0.4
    |    \--- software.amazon.awssdk:identity-spi:2.26.8
    |         +--- software.amazon.awssdk:annotations:2.26.8
    |         \--- software.amazon.awssdk:utils:2.26.8 (*)
    +--- software.amazon.awssdk:http-auth-aws:2.26.8
    |    +--- software.amazon.awssdk:annotations:2.26.8
    |    +--- software.amazon.awssdk:utils:2.26.8 (*)
    |    +--- software.amazon.awssdk:identity-spi:2.26.8 (*)
    |    +--- software.amazon.awssdk:http-client-spi:2.26.8 (*)
    |    +--- software.amazon.awssdk:http-auth-spi:2.26.8 (*)
    |    +--- software.amazon.awssdk:checksums-spi:2.26.8
    |    |    \--- software.amazon.awssdk:annotations:2.26.8
    |    \--- software.amazon.awssdk:checksums:2.26.8
    |         +--- software.amazon.awssdk:annotations:2.26.8
    |         \--- software.amazon.awssdk:checksums-spi:2.26.8 (*)
    +--- software.amazon.awssdk:checksums-spi:2.26.8 (*)
    +--- software.amazon.awssdk:checksums:2.26.8 (*)
    +--- software.amazon.awssdk:identity-spi:2.26.8 (*)
    +--- software.amazon.awssdk:utils:2.26.8 (*)
    +--- software.amazon.awssdk:profiles:2.26.8
    |    +--- software.amazon.awssdk:utils:2.26.8 (*)
    |    \--- software.amazon.awssdk:annotations:2.26.8
    +--- software.amazon.awssdk:retries-spi:2.26.8
    |    +--- software.amazon.awssdk:annotations:2.26.8
    |    \--- software.amazon.awssdk:utils:2.26.8 (*)
    +--- software.amazon.awssdk:retries:2.26.8
    |    +--- software.amazon.awssdk:retries-spi:2.26.8 (*)
    |    +--- software.amazon.awssdk:annotations:2.26.8
    |    \--- software.amazon.awssdk:utils:2.26.8 (*)
    \--- org.reactivestreams:reactive-streams:1.0.4
```

`software.amazon.awssdk:sts` has direct dependency on `software.amazon.awssdk:apache-client` and `software.amazon.awssdk:netty-nio-client` which kicks in all the `netty` related dependencies.


Despite the class `MSKCredentialProvider` doesn't directly refer to classes contained in `software.amazon.awssdk:sso` and `software.amazon.awssdk:ssooidc`, removing those doesn't provide any gain in terms of numerosity of dependencies.

So my conclusion is that we can't strip any jar.